### PR TITLE
Hotfix 5.1.0 - Fix adding multiple product to cart issue

### PR DESCRIPTION
**Pull request by olsi-qose**
## Description
Adding multiple products to cart is not working on magento cloud when the session is new. A request is made for each product added to cart.  When no quote is related to the session, multiple quotes with a single item will get created instead a single quote containing all products. The issue is fixed by chaining the request together.

Also a promise is returned after the execution. 

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] I have assigned the correct milestone or created one if non-existent.
- [ ] I have correctly labeled this pull request.
- [ ] I have linked the corresponding issue in this description.
- [ ] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [ ] I have checked the base branch of this pull request
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: view/frontend/web/js/recobuy.js [hunk 1]
```diff
@@ -35,56 +35,78 @@
...
-            products.forEach(function (productObj) {
-                Recobuy.addSkuToCart(productObj, element, 1);
-            });
+            return Recobuy.recursiveAddProducts(products, element)
+        } else {
+            Promise.reject(new Error("Products is not type array"))
         }
     };
 
+    Recobuy.recursiveAddProducts = function (products, element) {
```
Comment (mridang) [inline view/frontend/web/js/recobuy.js:64]: Maybe simplify? 

```
var readFiles = function(files) {
  return files.reduce((p, file) => {
     return p.then(() => readFile(file));
  }, Promise.resolve()); // initial
};
```

File: view/frontend/web/js/recobuy.js [hunk 2]
```diff
@@ -35,56 +35,78 @@
...
+                return Recobuy.recursiveAddProducts(products.slice(1), element)
+            })
+    }
+
     // Product object must have fields productId and skuId {'productId': '123', 'skuId': '321'}
-    Recobuy.addSkuToCart = function (product, element, quantity) {
-        quantity = quantity || 1;
+    Recobuy.addSkuToCart = function (product, element) {
+
+        var quantity = product.quantity || 1;
```
Comment (mridang) [inline view/frontend/web/js/recobuy.js:78]: Does this work? https://magento.stackexchange.com/questions/131189/magento-2-what-is-the-use-of-formkey

File: view/frontend/web/js/recobuy.js [hunk 3]
```diff
@@ -35,56 +35,78 @@
...
+
+        return new Promise(function (resolve, reject) {
+            $.post(url, {
+                form_key: formKey,
+                qty: quantity,
+                product: product.productId,
+                sku: product.skuId
+            }).done(function() {
+                Recobuy.sendCartEvent(element, product.productId)
+                return resolve()
```
Comment (mridang) [inline view/frontend/web/js/recobuy.js:91]: Do you need to return or simply resolve()?

File: view/frontend/web/js/recobuy.js [hunk 4]
```diff
@@ -35,56 +35,78 @@
...
+            quantity: quantity
         };
-        Recobuy.addSkuToCart(productData, element, quantity);
+        return Recobuy.addSkuToCart(productData, element);
     };
 
     // Products must be and array of objects [{'productId': '123', 'skuId': '321'}, {...}]
     // skuId is optional for simple products.
     Recobuy.addMultipleProductsToCart = function (products, element) {
         if (Array.isArray(products)) {
```
Comment (mridang) [inline view/frontend/web/js/recobuy.js:57]: I don't think we need this guard. We don't validate the object anyways. Remove the check

File: view/frontend/web/js/recobuy.js [hunk 5]
```diff
@@ -35,56 +35,78 @@
...
-    'catalogAddToCart'
+    'jquery'
 ], function (nostojs, $) {
     'use strict';
 
-    //noinspection SpellCheckingInspection
-    var form = $('#nosto_addtocart_form').catalogAddToCart({});
     var Recobuy = {};
+
     Recobuy.addProductToCart = function (productId, element, quantity) {
```
Comment (mridang) [inline view/frontend/web/js/recobuy.js:44]: Make quantity default https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

File: view/frontend/web/js/recobuy.js (hunk 2 above)
Comment (olsi-qose) [reply view/frontend/web/js/recobuy.js:78]: That's the way to add the form_key to the form. We use it here https://github.com/Nosto/nosto-magento2/blob/develop/view/frontend/templates/addtocart.phtml#L42

File: view/frontend/web/js/recobuy.js (hunk 3 above)
Comment (olsi-qose) [reply view/frontend/web/js/recobuy.js:91]: Return a promise

File: view/frontend/web/js/recobuy.js (hunk 4 above)
Comment (olsi-qose) [reply view/frontend/web/js/recobuy.js:57]: I can remove. It was altered here https://github.com/Nosto/nosto-magento2/pull/533/files#diff-f4a31733d3e7674b679d5cdfbff9eac23b3ba00542d9b44f98c1852d73625b9dR58

**Review by mridang — APPROVED** [review]

**Pull request merged**